src/explorer/LamarExtensions.cs

```diff
@@ -7,6 +7,13 @@ public static class LamarExtensions
     {
         public static T ResolvePublisherComponent<T>(this INestedContainer scope)
             where T : PublisherComponent
-        => (T)scope.GetInstance<PublisherComponent>(typeof(T).NameInCode());
+        {
+            // Try to resolve using the PublisherComponent interface. If this doesn't work, auto-resolve
+            // the concrete instance instead.
+            var fromCollection = (T)scope.TryGetInstance<PublisherComponent>(typeof(T).NameInCode());
+            return fromCollection is null
+                ? scope.GetInstance<T>()
+                : fromCollection;
```

Comment:
I s there any reason to not use the null coalescing operator?
```suggestion
          return fromCollection ?? scope.GetInstance<T>();
```